interfaces/cython/cantera/test/test_composite.py

```diff
@@ -569,6 +581,27 @@ def test_yaml_inconsistent_species(self):
         with self.assertRaisesRegex(ct.CanteraError, "different definitions"):
             gas.write_yaml('h2o2-error.yaml', phases=gas2)
 
+    def test_yaml_user_data(self):
+        gas = ct.Solution('h2o2.yaml')
+        extra = {"spam": {"A": 1, "B": 2}, "eggs": [1, 2.3, 4.5]}
+        gas.update_user_data(extra)
+        S = gas.species(2)
+        S.update_user_data({"foo": "bar"})
+        S.transport.update_user_data({"baz": 1234.5})
+        S.thermo.update_user_data({"something": (False, True)})
+        gas.reaction(5).update_user_data({"baked-beans": True})
+
+        gas.write_yaml("h2o2-generated-user-data.yaml")
+        gas2 = ct.Solution("h2o2-generated-user-data.yaml")
```

Comment:
This may be more of a feature request not appropriate to add here, but it'd be nice to be able to "write" the output into a `StringIO` buffer to avoid the filesystem entirely. Alternatively, to have a method that just returns a string of the generated YAML, which you can pass to the `yaml` kwarg of `Solution`.

Reply:
Given that the `write_yaml` method was introduced in #984 and not here, I think this is a separate feature request.